### PR TITLE
Adding Intermediate examples + Attribute examples

### PR DESCRIPTION
**Pull request by fungktt**
These intermediate examples are based on [these](http://docs.juliaplots.org/latest/examples/gr)

The attribute examples are not based on anything and are created to help other people understand what exactly the attributes do

### CODE REVIEW

**Comment by asinghvi17** [comment]
@fungktt could you move the examples to a Julia file with the cell/block syntax, as outlined [here](http://makie.juliaplots.org/dev/gallery.html#Structure-1)?  The examples themselves are great, thanks!

**Comment by fungktt** [comment]
@asinghvi17 Where should I place the Julia files?

**Comment by asinghvi17** [comment]
`examples/plots.jl` maybe?

File: examples/attributes.jl [hunk 1]
```diff
@@ -0,0 +1,131 @@
+@block KimFung [attribute] begin
```
Comment (asinghvi17) [inline examples/attributes.jl]:
```suggestion
@block KimFung ["attributes"] begin

File: examples/attributes.jl [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        scene = Scene()
+        scatter!(scene, rand(10), color=:red)
+        text!(scene,"adding text",textsize = 0.6, align = (:center, :center))
+        display(scene)
+    end
+
+    @cell "color" [lines, color] begin
+        vbox(
+            lines(rand(10), linewidth = 20, color = :blue)
+            lines(rand(10), linewidth = 20, color = to_colormap(:viridis, 10))
```
Comment (asinghvi17) [inline examples/attributes.jl]: Maybe a comment here about what this does?  You could also just give `color` as `1:10`, which should do the same thing.

File: examples/attributes.jl [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        )
+    end
+
+    @cell "colormap" [lines, colormap] begin
+        t = range(0, stop=1, length=300)
+        θ = (6π) .* t
+        x = t .* cos.(θ)
+        y = t .* sin.(θ)
+        lines(x,y,color=t,colormap=:colorwheel,linewidth=8,scale_plot= false)
```
Comment (asinghvi17) [inline examples/attributes.jl]:
```suggestion
        lines(x, y, color = t, colormap = :colorwheel, linewidth = 8, scale_plot = false)
```

File: examples/attributes.jl [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    end
+
+    @cell "colorrange" [lines, colormap, colorrange] begin
+        lines(randn(10),color=LinRange(-1, 1, 10),colormap=:colorwheel,linewidth=8, colorrange = (-1.0,1.0))
+    end
+
+    @cell "fillrange" [lines, colormap, colorrange] begin
```
Comment (asinghvi17) [inline examples/attributes.jl]:
```suggestion
    @cell "fillrange" [contour, fillrange] begin

File: examples/attributes.jl [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        y = LinRange(-1, 1, 20)
+        z = x .* y'
+        contour(x, y, z, levels = 0, linewidth = 0, fillrange = true)
+    end
+
+    @cell "font" [text, scatter, font] begin
+        scene = Scene()
+        scatter!(scene, rand(10), color=:red)
+        text!(scene,"adding text",textsize = 0.6, align = (:center, :center), font = "Blackchancery")
+        display(scene)
```
Comment (asinghvi17) [inline examples/attributes.jl]:
```suggestion

File: examples/attributes.jl [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    end
+
+    @cell "glowcolor, glowwidth" [scatter, glowcolor, glowwidth] begin
+        scatter(randn(10),color=:blue, glowcolor = :orange, glowwidth = 10)
+    end
+
+    @cell "glowcolor, glowwidth" [scatter, glowcolor, glowwidth] begin
```
Comment (asinghvi17) [inline examples/attributes.jl]: Duplicated?  See above

File: examples/attributes.jl [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+        scatter(randn(10),color=:blue, glowcolor = :orange, glowwidth = 10)
+    end
+
+    @cell "image" [image] begin
+        import FileIO
+        img = FileIO.load(download("http://makie.juliaplots.org/dev/assets/logo.png"))
```
Comment (asinghvi17) [inline examples/attributes.jl]:
```suggestion
        img = Makie.logo()
```

File: examples/attributes.jl [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    @cell "strokecolor, strokewidth" [poly, stokecolor, strokewidth] begin
+        x = LinRange(0, 2pi, 100)
+        poly(Point2f0.(zip(sin.(x), sin.(2x))), color = :white, strokecolor = :blue, strokewidth = 10)
+    end
+
+    @cell "textsize" [text, scatter, textsize] begin
+        scene = Scene()
+        scatter!(scene, rand(10), color = to_colormap(:colorwheel, 10))
+        text!(scene, "hello world", textsize = 2)
+        display(scene)
```
Comment (asinghvi17) [inline examples/attributes.jl]:
```suggestion

File: examples/attributes.jl [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+    end
+
+    @cell "visible" [scatter, visible] begin
```
Comment (asinghvi17) [inline examples/attributes.jl]: I think this already has an example, with a stepper.  Maybe something showing a Scene, where one plot is made invisible would be better?

File: examples/intermediate.jl [hunk 10]
```diff
@@ -0,0 +1,141 @@
+@block KimFung [intermediate] begin
+
+    @cell "simple line plot" [lines] begin
+        scene = Scene()
+        lines!(scene, rand(10), linewidth=3.0, color=:blue)
+        lines!(scene, rand(10), linewidth=3.0, color=:red)
+        lines!(scene, rand(10), linewidth=3.0, color=:green)
+        lines!(scene, rand(10), linewidth=3.0, color=:purple)
+        lines!(scene, rand(10), linewidth=3.0, color=:orange)
+        display(scene)
```
Comment (asinghvi17) [inline examples/intermediate.jl]:
```suggestion

File: examples/intermediate.jl [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+    end
+
+    @cell "multiple functions" [lines] begin
+        scene = Scene()
+        x = range(0, stop = 3pi, step = 0.01)
+        lines!(scene, x, sin.(x), color = :black)
+        lines!(scene, x, cos.(x), color = :blue)
+        display(scene)
```
Comment (asinghvi17) [inline examples/intermediate.jl]:
```suggestion

File: examples/intermediate.jl [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+    end
+
+    @cell "animations" [lines] begin
+        scene = Scene()
+        mytime = Node(0.0)
+        f(v, t) = sin(v + t)
+        g(v, t) = cos(v + t)
+        lines!(scene,lift(t -> f.(range(0, stop = 2pi, length = 50), t), mytime),color = :blue)
+        lines!(scene,lift(t -> g.(range(0, stop = 2pi, length = 50), t), mytime),color = :orange)
+        record(scene, "mygraph.gif", range(0, stop = 4pi, length = 100)) do i
```
Comment (asinghvi17) [inline examples/intermediate.jl]:
```suggestion
        record(scene, @replace_with_a_path(mp4), range(0, stop = 4pi, length = 100)) do i

File: examples/intermediate.jl [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+            mytime[] = i
+        end
+    end
+
+    @cell "parametric plots" [poly] begin
+        x = LinRange(0, 2pi, 100)
+        poly(Point2f0.(zip(sin.(x), sin.(2x))), color = "orange", strokecolor = "blue", strokewidth = 4)
```
Comment (asinghvi17) [inline examples/intermediate.jl]:
```suggestion
        poly(Point2f0.(sin.(x), sin.(2x)), color = "orange", strokecolor = "blue", strokewidth = 4)

File: examples/intermediate.jl [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+    end
+
+    @cell "colors" [lines] begin
+        lines(1:50, 1:1, linewidth = 20, color = to_colormap(:viridis, 50))
+    end
+
+    @cell "image" [image] begin
+        import FileIO
+        img = FileIO.load(download("http://makie.juliaplots.org/dev/assets/logo.png"))
```
Comment (asinghvi17) [inline examples/intermediate.jl]:
```suggestion
        img = Makie.logo()

File: examples/intermediate.jl [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+        image(img, scale_plot = false)
+    end
+
+    @cell "attributes" [attributes, lines, color, colormap, linewidth] begin
+        scene = Scene()
+        lines!(scene, rand(10), color = to_colormap(:viridis, 10), linewidth = 5)
+        lines!(scene, rand(20), color = :red, alpha = 0.5)
+        display(scene)
```
Comment (asinghvi17) [inline examples/intermediate.jl]:
```suggestion

File: examples/intermediate.jl [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+    end
+
+    @cell "build plot in pieces" [lines] begin
```
Comment (asinghvi17) [inline examples/intermediate.jl]: Better to use a [Stepper](http://makie.juliaplots.org/dev/output.html#Stepper-plots-1) here

File: examples/intermediate.jl [hunk 17]
```diff
@@ -0,0 +1,141 @@
...
+        scene1, scene2, scene3, scene4 = Scene(), Scene(), Scene(), Scene()
+        lines!(scene1, rand(10), color=:red)
+        lines!(scene1, rand(10), color=:blue)
+        lines!(scene1, rand(10), color=:green)
+        scatter!(scene2, rand(10), color=:red)
+        scatter!(scene2, rand(10), color=:blue)
+        scatter!(scene2, rand(10), color=:orange)
+        barplot!(scene3, randn(99))
+        plot!(scene4, histogram, randn(1000))
+        display(vbox(hbox(scene2,scene1),hbox(scene4,scene3)))
```
Comment (asinghvi17) [inline examples/intermediate.jl]:
```suggestion
        vbox(hbox(scene2,scene1),hbox(scene4,scene3))

File: examples/intermediate.jl [hunk 18]
```diff
@@ -0,0 +1,141 @@
...
+    end
+
+    @cell "text" [scatter, text] begin
+        scene = Scene()
+        scatter!(scene, rand(10), color=:red)
+        text!(scene,"adding text",textsize = 0.6, position = (5.0, 1.1))
+        display(scene)
```
Comment (asinghvi17) [inline examples/intermediate.jl]:
```suggestion

Comment (asinghvi17) [inline examples/intermediate.jl]:
```suggestion

File: examples/intermediate.jl [hunk 19]
```diff
@@ -0,0 +1,141 @@
...
+        heatmap(x, y, z)
+    end
+
+    @cell "label rotation, title location" [lines, axisnames, rotation, legend, position, vbox] begin
+        scene = Scene()
+        line = lines!(scene, rand(10), linewidth=3.0, color=:red)
+        scene[Axis][:names, :axisnames] = ("text","")
+        scene[Axis][:names, :rotation] = (1.5pi)
+        leg = legend([scene[2]], ["line"], position = (1, -1))
+        display(vbox(line,leg))
```
Comment (asinghvi17) [inline examples/intermediate.jl]:
```suggestion
        vbox(line, leg)

File: examples/intermediate.jl [hunk 20]
```diff
@@ -0,0 +1,141 @@
...
+    end
+
+    @cell "lines with varying colors" [lines, scatter, vbox, colormap] begin
+        t = range(0, stop=1, length=100)
+        θ = (6π) .* t
+        x = t .* cos.(θ)
+        y = t .* sin.(θ)
+        p1 = lines(x,y,color=t,colormap=:colorwheel,linewidth= 8,scale_plot= false)
+        p2 = scatter(x,y,color=t,colormap=:colorwheel,linewidth= 8,scale_plot= false)
+        display(vbox(p1, p2))
```
Comment (asinghvi17) [inline examples/intermediate.jl]:
```suggestion
        vbox(p1, p2)
```

File: examples/attributes.jl (hunk 2 above)
Comment (fungktt) [reply examples/attributes.jl]: Done

File: examples/attributes.jl (hunk 6 above)
Comment (fungktt) [reply examples/attributes.jl]: Done

File: examples/attributes.jl (hunk 9 above)
Comment (fungktt) [reply examples/attributes.jl]: Done

File: examples/intermediate.jl (hunk 16 above)
Comment (fungktt) [reply examples/intermediate.jl]: Done!

File: examples/intermediate.jl [hunk 21]
```diff
@@ -51,11 +51,13 @@
...
 
     @cell "build plot in pieces" [lines] begin
-        scene = Scene()
-        lines!(scene, rand(50)/3, color = :purple, linewidth = 5)
-        display(scene)
-        scatter!(scene, rand(50), color = :orange, markersize = 1)
-        display(scene)
+         scene = Scene()
+         # initialize the stepper and give it an output destination
+         st = Stepper(scene, "mystepper")
```
Comment (asinghvi17) [inline examples/intermediate.jl]:
```sugggestion
         st = Stepper(scene, @replace_with_a_path)

Comment (fungktt) [reply examples/intermediate.jl]: Done!

**Pull request merged**